File: app/src/main/java/br/com/samueltobias/customercrud/database/dao/CustomerDao.java
package br.com.samueltobias.customercrud.database.dao;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;

import java.util.List;

import br.com.samueltobias.customercrud.model.Customer;

@Dao
public interface CustomerDao {
    @Insert
    void save(Customer customer);

    @Query(value = "SELECT * FROM customer")
    List<Customer> getAll();
}

File: app/src/main/java/br/com/samueltobias/customercrud/database/dao/CustomerDao.kt
package br.com.samueltobias.customercrud.database.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import br.com.samueltobias.customercrud.model.Customer

@Dao
interface CustomerDao {
    @Insert
    fun save(customer: Customer)

    @Query(value = "SELECT * FROM customer")
    fun getAll(): List<Customer>

    @Query(value = "SELECT * FROM customer WHERE id = :id")
    fun get(id: Long): Customer?
}